### PR TITLE
Adjust to number of captures in previous regex

### PR DESCRIPTION
**Pull request by mgautierfr**
supersedes #359

### CODE REVIEW

File: test/zimwriterfs-zimcreatorfs.cpp [hunk 1]
```diff
@@ -127,3 +128,49 @@ TEST(ZimCreatorFSTest, ThrowsErrorIfDirectoryNotExist)
     ZimCreatorFS zimCreator("Non-existing-dir");
   }, std::invalid_argument );
 }
+
+struct Redirect {
+  std::string path;
+  std::string title;
+  std::string target;
+};
```
Comment (veloman-yunkan) [inline test/zimwriterfs-zimcreatorfs.cpp:136]: Why don't you put this in `zimcreatorfs.h` and replace the three parameters of the redirect handler with a single one? It also makes sense to declare `parse_redirectArticles()` in `zimcreatorfs.h` too.

File: test/zimwriterfs-zimcreatorfs.cpp [hunk 2]
```diff
@@ -127,3 +128,49 @@ TEST(ZimCreatorFSTest, ThrowsErrorIfDirectoryNotExist)
...
+  std::vector<Redirect> found;
+  parse_redirectArticles(
+    ss,
+    [&](std::string path, std::string title, std::string target)
+     {found.push_back({path, title, target});}
+  );
+
+  std::vector<Redirect> expected;
+  expected.push_back({"path", "title", "target"});
+  expected.push_back({"A/path/to/somewhere", "An amazing title", "Another/path"});
```
Comment (veloman-yunkan) [inline test/zimwriterfs-zimcreatorfs.cpp:160]: IMHO, using an initializer list would make this piece of code slightly more readable (as well as will allow to declare this variable `const`).

File: test/zimwriterfs-zimcreatorfs.cpp [hunk 3]
```diff
@@ -127,3 +128,49 @@ TEST(ZimCreatorFSTest, ThrowsErrorIfDirectoryNotExist)
...
+    std::stringstream ss;
+    ss << "A/path\tOups, no target";
+    EXPECT_THROW({
+      parse_redirectArticles(
+            ss,
+            [&](std::string path, std::string title, std::string target)
+             {}
+          );
+    }, std::runtime_error);
+  }
```
Comment (veloman-yunkan) [inline test/zimwriterfs-zimcreatorfs.cpp:175]: Please also add a test with more than 3 fields (more than 2 tab separators).

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]

File: test/zimwriterfs-zimcreatorfs.cpp (hunk 1 above)
Comment (mgautierfr) [reply test/zimwriterfs-zimcreatorfs.cpp]: Done

File: test/zimwriterfs-zimcreatorfs.cpp (hunk 2 above)
Comment (mgautierfr) [reply test/zimwriterfs-zimcreatorfs.cpp]: Done

File: test/zimwriterfs-zimcreatorfs.cpp (hunk 3 above)
Comment (mgautierfr) [reply test/zimwriterfs-zimcreatorfs.cpp]: Done.
And, there was a bug here. Now fixed.

File: test/zimwriterfs-zimcreatorfs.cpp [hunk 4]
```diff
@@ -165,4 +165,16 @@ TEST(ZimCreatorFSTest, ParseRedirect)
           );
     }, std::runtime_error);
   }
+
+  {
+      std::stringstream ss;
+      ss << "A/path\ttitle\ttarget\tOups, to many tabs\n";
```
Comment (veloman-yunkan) [inline test/zimwriterfs-zimcreatorfs.cpp:171]: to many -> too many

**Review by veloman-yunkan — CHANGES_REQUESTED** [review]
LGTM.

Please rebase&fixup&approve&merge

**Comment by mgautierfr** [comment]
@veloman-yunkan comment fixed.
Packages CI failing is not related to this change.

I'm merging.

**Pull request merged**